Source/WebCore/loader/HistoryController.h

```diff
@@ -45,7 +45,7 @@ enum class ShouldTreatAsContinuingLoad : uint8_t;
 struct NavigationAPIMethodTracker;
 struct StringWithDirection;
 
-class HistoryController final : public CanMakeCheckedPtr<HistoryController> {
+class HistoryController final : public CanMakeCheckedPtr<HistoryController>, public CanMakeWeakPtr<HistoryController> {
```

Comment:
I see you still did not forward the ref-counting, I think we really should. Normally you'd get a build error if you subclass `CanMakeWeakPtr` without being ref-counted. However, the `CanMakeCheckedPtr` is saving you before in theory, it would be safe (from a security standpoint) to convert the WeakPtr into a CheckedPtr and then only use that CheckedPtr. However, your lambdas just use the WeakPtr directly and never convert to a CheckedPtr or a RefPtr so the code is not safe. I do think that RefPtr would be better than CheckedPtr given that it is trivial to forward the ref-counting here (I think?) and it would avoid crashes if the object gets destroyed in the middle of the lambda running.

Reply:
I mentioned this to you on Slack, actually - I cannot forward ref counting without completely reworking a fragile class hierarchy, because HistoryController is currently a UniqueRef, which is incompatible with RefCounted.

Reply:
On Slack we established the build break, and discovered the workaround of makeUniqueRefWithoutRefCountedCheck

Reply:
calling `makeUniqueRefWithoutRefCountedCheck` would resolve your build issue when forwarding refcounting and using UniqueRef.